pkg/deploy/kubernetes_image_puller.go

```diff
@@ -35,6 +34,41 @@ import (
 
 var imagePullerFinalizerName = "kubernetesimagepullers.finalizers.che.eclipse.org"
 
+type ImageKeyValue struct {
+	key   string
+	value string
+}
+
+// GetImageRepoName returns the image repo name of the provided image
+// Examples:
+// GetImageRepoName("quay.io/test/test:tag") returns "quay.io/test/test"
+// GetImageRepoName("quay.io/test/test@sha256:abcdef") returns "quay.io/test/test"
+func GetImageRepoName(image string) string {
+	repoName := image
+	colonIndex := strings.Index(image, ":")
+	if colonIndex >= 0 {
+		repoName = repoName[:colonIndex]
+	}
+
+	atIndex := strings.Index(repoName, "@")
+	if atIndex >= 0 {
+		repoName = repoName[:atIndex]
+	}
+	return repoName
+}
+
+// GetImageTag returns the tag of the provided image
+// Examples:
+// GetImageTag("quay.io/test/test:tag") returns "tag"
+// GetImageTag("quay.io/test/test@sha256:abcdef") returns "abcdef"
+func GetImageTag(image string) string {
+	lastIndex := strings.LastIndex(image, ":")
+	if lastIndex >= 0 {
+		return image[lastIndex+1:]
+	}
+	return ""
+}
+
```

Comment:
Pls
1. Combine `GetImageRepoName` and `GetImageTag` into a single function returning (string, string).
2. Right now they are fluky, have a look at the similar one [1]
3. Add tests,

[1] https://github.com/che-incubator/chectl/blob/main/src/util.ts#L93-L124

Reply:
thank you, fixed